content/800-data-platform/700-schema-viewer.mdx

```diff
@@ -63,9 +63,34 @@ You can use the **Refresh Schema** option only if you are the [Project owner](/d
   alt="Schema Viewer - Projects with introspected Prisma schema"
 />
 
+### Upload a schema file
+
+If the schema refresh fails, you can upload a schema file, an error notification pops up at the top of the page with the option to upload a schema file.
+
+Click the **upload a schema** link in the notification. Below the **Provide a schema** section, a new **Upload a Prisma schema file** section is enabled. You can either:
+
+- Click on the _drop zone_ to open your file manager and select your schema
+- Drag and drop your schema file to the _drop zone_
+
+Review the contents of the schema file in the _drop zone_ and once you are done, click **Upload file**
+
+<Admonition>
+
+**Note**<br /><br />
+
+The schema preview does not contain any syntax highlighting for your schema file.
+
+</Admonition>
+
+<img
+  src="images/schema-viewer/project-schema-viewer.png"
+  title="Schema Viewer - Projects with no models and no Prisma schema"
+  alt="Schema Viewer - Projects with no models and no Prisma schema"
+/>
+
 ## Projects without a schema
 
-If you create your project with an empty database (one which contains no models) and do not enable schema synchronization, the platform tries to introspect the database but because that contains no models, the platform cannot generate a Prisma schema for it.
+If you create your project with an empty database (one which contains no models), or no schema (one which introspection failed, and/ or uploaded a file), or and do not enable schema synchronization, the platform tries to introspect the database but because that contains no models, the platform cannot generate a Prisma schema for it.
```

Comment:
Hmm we could work on this chapter a bit. What is the intended message we try to deliver here?

Some details to guide us:
- If database has prisma schema, but no data, introspection can succeed
- if database has some data, but no existing prisma schema, introspection can succeed
- If database has no schema and is empty, introspection fails

Maybe we could rephrase this chapter something like:

```
If introspection failed during project creation and you chose to skip providing a schema, your project exists without a schema associated to it.

For such projects, [**Data Browser**](/data-platform/data-browser) and [**Query Console**](/data-platform/query-console) remain disabled until you enable schema synchronization or manually re-trigger a new introspection.
```

Note that I on purpose left out the "uploda a Prisma schema". Maybe we do not need it here as it is described above (if refresh fails).

WDYT?

Reply:
Thanks for the suggestion. I'll include it.